mappings/net/minecraft/entity/Entity.mapping

```diff
@@ -573,7 +573,17 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
 	METHOD method_5721 isTouchingWaterOrRain ()Z
 	METHOD method_5722 isTeammate (Lnet/minecraft/class_1297;)Z
 		ARG 1 other
-	METHOD method_5723 dealDamage (Lnet/minecraft/class_1309;Lnet/minecraft/class_1297;)V
+	METHOD method_5723 applyDamageEffects (Lnet/minecraft/class_1309;Lnet/minecraft/class_1297;)V
+		COMMENT Applies damage effects to {@code attacker} or {@code target}.
+		COMMENT
+		COMMENT <p>Called when {@code attacker} damages {@code target}. The attacker is
+		COMMENT usually this entity, but may be a projectile's owner entity.
```

Comment:
Would other deferred damage sources such as TNT not apply here?

Reply:
Only projectiles. TNT doesn't track its owner.

Reply:
This is controlled by the entity code that calls this method. Worthy of documentation?

Reply:
TNT does track its owner:

![Player661 was blown up by Player661](https://user-images.githubusercontent.com/24855774/120073004-7eb53700-c064-11eb-9cef-a78e82ab1301.png)

Is there a specific method that controls what the `attacker` is? I would document that here.

Reply:
Don't think so; I think that was controlled by damage source. This is more about enchantments.

Another fun fact I just discovered is that `PersistentProjectileEntity` calls the `onUserDamaged` and `onTargetDamaged` methods individually in `onEntityHit(EntityHitResult)`. :zany_face: please mojang

Reply:
> TNT does track its owner

You're right. I meant it doesn't track for enchantment effect purposes.

> Is there a specific method that controls what the `attacker` is? I would document that here.

`attacker` is either this entity or `ProjectileEntity#getOwner()` . I don't see any specific method for it.

Reply:
> I don't see any specific method for it.

Hmm, they added a method for status effect cause. Hope they will use that method for both purposes soon :tm:
https://github.com/FabricMC/yarn/pull/2441/files#diff-0583232f2335075f834f611a33196ed09274d74e21f20a4c1c99ae70ff69abcdR25